### PR TITLE
holdingPieceLedSignals

### PR DESCRIPTION
**Pull request by Anuva12**

### CODE REVIEW

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 1]
```diff
@@ -72,12 +75,20 @@ public class LEDManagerSubsystem extends SubsystemBase {
 
     private final MirroredLEDFlash m_isInLoadingZoneSignal;
 
-    public LEDManagerSubsystem(ChassisSubsystem chassisSubsystem, ArmPivotSubsystem armSubsystem, TurretSubsystem turretSubsystem, AutonomousFactory autonomousFactory) {
+    private final MirroredLEDFlash m_isHoldingPieceClaw;
+
+    private final Timer m_timer = new Timer();
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:80]: Timer for what?

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 2]
```diff
@@ -165,6 +180,11 @@ private void enabledPatterns() {
         else if (m_optionCubeLED) {
             m_cubeGamePieceSignal.writeLeds();
         }
+        else if (m_claw.hasGamePiece() && m_timer.get() < 1) {
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:183]: Make the time a constant

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 3]
```diff
@@ -165,6 +180,11 @@ private void enabledPatterns() {
...
+            System.out.println("claw has piece");
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:184]: Remove debug

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 4]
```diff
@@ -190,16 +211,27 @@ else if (m_optionDockLED) {
         */
     }
 
+    public void shouldTrip() {
+        if (!m_clawWasTripped && m_claw.hasGamePiece()) {
+            m_timer.reset();
+            m_timer.start();
+            System.out.println("Claw Timer has reset");
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:218]: Remove debug

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 5]
```diff
@@ -190,16 +211,27 @@ else if (m_optionDockLED) {
...
         clear();
-
         if (DriverStation.isDisabled()) {
             disabledPatterns();
         }
         else {
             enabledPatterns();
         }
+        shouldTrip();
+        m_clawWasTripped = m_claw.hasGamePiece();
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:233]: Now that `shouldTrip` is also getting called in periodic, you can probbaly move `m_clawWasTripped = m_claw.hasGamePiece();`  back into there.

I don't like that name either

File: y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java [hunk 6]
```diff
@@ -137,6 +148,10 @@ public LEDManagerSubsystem(ChassisSubsystem chassisSubsystem, ArmPivotSubsystem
         //for Claw Aligned Check
         m_clawAlignedSignal = new MirroredLEDFlash(m_buffer, 0, MAX_INDEX_LED, 0.5, Color.kOrange);
 
+        //holding piece in claw
+        m_isHoldingPieceClaw = new MirroredLEDFlash(m_buffer, 0, MAX_INDEX_LED, 0.5, Color.kRed);
```
Comment (pjreiniger) [inline y2023/ChargedUp/src/main/java/com/gos/chargedup/subsystems/LEDManagerSubsystem.java:152]: Make this flash faster, like 10x a second instead of once

**Review by pjreiniger — CHANGES_REQUESTED** [review]

**Review by ashley-wei — APPROVED** [review]

**Review by pjreiniger — APPROVED** [review]

**Pull request merged**